### PR TITLE
Fix ramsey uuid denormalization #3473

### LINKED ISSUE #3473: Exception when using Ramsey/Uuid ^4.0
**API Platform version(s) affected**: x.y.z

**Description**  
After updating to **Ramsey/Uuid ^4.0** i get an exception when querying an item with identifier. For example  api/books/3fa85f64-5717-4562-b3fc-2c963f66afa6

The exception:
`Argument 1 passed to Ramsey\Uuid\Uuid::fromString() must be of the type string, object given, called in ..\vendor\api-platform\core\src\Bridge\RamseyUuid\Identifier\Normalizer\UuidNormalizer.php on line 35`

$data contains an Uuid object instead of an string , the Uuid is my identifier.

It works fine with **Ramsey/Uuid ^3.9** !

**Possible Solution**  
Add a check if $data is a string or an instance of UuidInterface in the UuidNormalizer

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #3473 
| License       | MIT
| Doc PR        | todo

### CODE REVIEW

File: src/Bridge/RamseyUuid/Serializer/UuidDenormalizer.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\RamseyUuid\Serializer;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Component\Serializer\Normalizer\ContextAwareDenormalizerInterface;
+
+class UuidDenormalizer implements ContextAwareDenormalizerInterface
```
Comment (norkunas) [inline src/Bridge/RamseyUuid/Serializer/UuidDenormalizer.php:20]: Why not just `DenormalizerInterface`?

Comment (soyuka) [reply src/Bridge/RamseyUuid/Serializer/UuidDenormalizer.php]: Yeah why not :p also fixed 7.1 issue thx.

File: src/Bridge/RamseyUuid/Serializer/UuidDenormalizer.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\RamseyUuid\Serializer;
+
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class UuidDenormalizer implements DenormalizerInterface
```
Comment (dunglas) [inline src/Bridge/RamseyUuid/Serializer/UuidDenormalizer.php:20]:
```suggestion
final class UuidDenormalizer implements DenormalizerInterface
```

**Comment by soyuka** [comment]
see https://github.com/api-platform/core/pull/3511 cause of cache issue

**Pull request closed without merging**